tests/test_raft.rs

```diff
@@ -1923,3 +1931,291 @@ fn test_commit_after_remove_node() {
     assert_eq!(ents[0].get_entry_type(), EntryType::EntryNormal);
     assert_eq!(ents[0].get_data(), "hello".as_bytes());
 }
+
+// test_leader_transfer_to_uptodate_node verifies transferring should succeed
+// if the transferee has the most up-to-date log entires when transfer starts.
+#[test]
+fn test_leader_transfer_to_uptodate_node() {
+    let mut nt = Network::new(vec![None, None, None]);
+    nt.send(vec![new_message(1, 1, MessageType::MsgHup, 0)]);
+
+    let lead_id = nt.peers[&1].leader_id;
```

Comment:
no...in rust... we can't.
if lead is borrowed here, later when we call check_leader_transfer_state, compiler would
complain can't borrow mutable more than once.